Std/Data/Int/Lemmas.lean

```diff
@@ -841,6 +841,29 @@ protected theorem add_neg_of_nonpos_of_neg {a b : Int} (ha : a ≤ 0) (hb : b <
 protected theorem lt_add_of_le_of_pos {a b c : Int} (hbc : b ≤ c) (ha : 0 < a) : b < c + a :=
   Int.add_zero b ▸ Int.add_lt_add_of_le_of_lt hbc ha
 
+protected theorem add_le_iff_le_sub (a b c : Int) : a + b ≤ c ↔ a ≤ c - b := by
+  conv =>
+    lhs
+    rw [← Int.add_zero c, ← Int.sub_self (-b), Int.sub_eq_add_neg, ← Int.add_assoc, Int.neg_neg,
+      Int.add_le_add_iff_right]
+
+protected theorem le_add_iff_sub_le (a b c : Int) : a ≤ b + c ↔ a - c ≤ b := by
+  conv =>
+    lhs
+    rw [← Int.neg_neg c, ← Int.sub_eq_add_neg, ← Int.add_le_iff_le_sub]
+
+protected theorem add_le_zero_iff_le_neg (a b : Int) : a + b ≤ 0 ↔ a ≤ - b := by
+  rw [Int.add_le_iff_le_sub, Int.zero_sub]
+
+protected theorem add_le_zero_iff_le_neg' (a b : Int) : a + b ≤ 0 ↔ b ≤ -a := by
+  rw [Int.add_comm, Int.add_le_zero_iff_le_neg]
+
+protected theorem add_nonnneg_iff_neg_le (a b : Int) : 0 ≤ a + b ↔ -b ≤ a := by
+  rw [Int.le_add_iff_sub_le, Int.zero_sub]
+
+protected theorem add_nonnneg_iff_neg_le' (a b : Int) : 0 ≤ a + b ↔ -a ≤ b := by
+  rw [Int.add_comm, Int.add_nonnneg_iff_neg_le]
```

Comment:
I know these aren't `simp`, but I am surprised by the direction of these `iff` rules, I would have expected them to normalize the expressions in some way perhaps to `0 <= x + ... + y`.

Could you say a little about why the orientation of the left and right-hand sides?

Reply:
Just the direction these turned up:
```
w: c + Coeffs.dot x y ≤ 0 ∧ 0 ≤ c + Coeffs.dot x y
⊢ -c ≤ Coeffs.dot x y ∧ Coeffs.dot x y ≤ -c
```
No objection to flipping them, or making them `omega` private.

I'm a little skeptical that there's any general sense in which one side is simpler than the other (possibly why these lemmas don't already exist).

Reply:
In that case, I'd just be inclined to keep them with omega for now.